### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by cunninghamge**
Making one more attempt on this branch before I blow it up and make a new one. The most recent change I made was to move the 'travis' gem from the development group to the development & test group - which I had removed it from earlier because rubocop pointed out that we had it in both places but I think I removed it from the wrong one. I also changed our settings on Travis so that they are the same as the ones 
that Adam and Joe both said they had used successfully: I turned 'Build pushed branches' back on and I turned both of the 'Auto Cancellation' settings off. Hopefully that will allow us to deploy.

Neccesary checkmarks:

    [x] All Tests are Passing

    [x] The code will run locally

Type of change

    [x] New feature
    [] Bug Fix

Testing Changes

    [] No Tests have been changed
    [x] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

    [] My code has no unused/commented out code
    [] I have reviewed my code
    [] I have commented my code, particularly in hard-to-understand areas
    [] I have fully tested my code

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      flash.now[:errors] = @user.errors.full_messages
+      render :new, obj: @user
+    end
+  end
+
+  private
+
+  def user_params
+    user_info = params.require(:user).permit(:name, :email, :password, :password_confirmation)
+    user_info[:email] = user_info[:email].downcase
```
Comment (isabellevillasenor) [inline app/controllers/users_controller.rb:21]: Looks great! Can we move `.downcase` into the model?

File: app/views/shared/_user_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<%= form_with model: @user, local: true do |f| %>
+  <%= f.label :name if confirmation %>
```
Comment (isabellevillasenor) [inline app/views/shared/_user_form.html.erb:2]: I love the work Gus! Can we rename 'confirmation' to 'new_user', or something of the sorts?

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<h1>Welcome to Viewing Party</h1>
+<p>Please Register for an Account</p>
+<%= render partial: 'shared/user_form', locals: {button_text: 'Register', confirmation: true} %>
```
Comment (isabellevillasenor) [inline app/views/users/new.html.erb:3]: Lil' flag

File: app/views/welcome/index.html.erb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <h1>Welcome to Viewing Party!</h1>
 <p>An application to explore movies and create a viewing party event for you and your friends to watch a movie together</p>
 
-<%= button_to 'Log In' %>
-<%= link_to 'Register' %>
+<%= render partial: 'shared/user_form', locals: {button_text: 'Log In', confirmation: false} %>
```
Comment (isabellevillasenor) [inline app/views/welcome/index.html.erb:4]: Another lil' flag

**Review by isabellevillasenor — COMMENTED** [review]
Thanks again for all the work Gus! And for dealing with Travis >:[

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (cunninghamge) [reply app/controllers/users_controller.rb:21]: yes!

**Pull request merged**